src/components/common/Notification/NotificationBar.vue

```diff
@@ -0,0 +1,248 @@
+<template>
+  <div
+    :class="[
+      'noti',
+      alertType,
+      'animate__animated',
+      show && 'show',
+      show ? 'animate__fadeInRight' : 'animate__fadeOutRight',
+    ]"
+  >
+    <div class="noti-content" @mouseover="showCloseBtn = true">
+      <div class="row--close">
+        <div class="column--title">
+          <span v-if="isSuccessType || isCopiedType" class="icon--check">
+            <astar-icon-circle-check size="20" />
+          </span>
+          <span v-else class="icon--check">
+            <astar-icon-warning size="20" />
+          </span>
+          {{ alertTypeTitle }}
+        </div>
+        <div v-if="showCloseBtn" class="column--close" @click="close">
+          <astar-icon-close />
+        </div>
+      </div>
+      <div v-if="!isCopiedType">
+        <div class="message">{{ alertMsg }}</div>
+        <astar-button v-if="isSuccessType && txHash" class="btn--check" @click="goToSubscan">{{
+          $t('toast.checkYourTransactions')
+        }}</astar-button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { AlertType } from 'src/store/general/state';
+import { useNetworkInfo } from 'src/hooks';
+import { endpointKey } from 'src/config/chainEndpoints';
+import { useI18n } from 'vue-i18n';
+import { defineComponent, toRefs, PropType, computed, ref } from 'vue';
+
+export default defineComponent({
+  name: 'NotificationBar',
+  props: {
+    show: {
+      type: Boolean,
+      default: false,
+    },
+    alertType: {
+      type: String as PropType<AlertType>,
+      required: true,
+    },
+    alertMsg: {
+      type: String,
+      required: true,
+    },
+    txHash: {
+      type: String,
+      default: null,
+    },
+  },
+  emits: ['close'],
+  setup(props, { emit }) {
+    const close = () => {
+      emit('close');
+    };
+
+    const { t } = useI18n();
+
+    const alertTypeTitle = computed(() => {
```

Comment:
just a small thing, let's add a type annotation to `ref` and `computed`

Reply:
Fixed in [here](https://github.com/AstarNetwork/astar-apps/pull/644/commits/c6d329b5df7c4c973a86ec15db606afe352dc81d)